clippy_lints/src/loops/mod.rs

```diff
@@ -379,9 +379,9 @@ declare_clippy_lint! {
     /// ### Example
     /// ```no_run
     /// let mut foo = 42;
-    /// for i in 0..foo {
+    /// for i in 0..=foo {
     ///     foo -= 1;
-    ///     println!("{}", i); // prints numbers from 0 to 42, not 0 to 21
+    ///     println!("{i}"); // prints numbers from 0 to 42, not 0 to 41
```

Comment:
I think that the `21` in the comment was right. The way I'm understanding this example is that you're decrementing the high part of the range on *every iteration*, so the expectation would be that it when `i` is 21, `foo` should have been decremented down to `20` and the loop would break there.

Reply:
Thanks, @y21, you're quite right. I was just being a bit slow-witted here.